src/codeu/chat/client/android/app/src/main/java/com/google/codeu/chatme/controller/Controller.java

```diff
@@ -0,0 +1,74 @@
+package com.google.codeu.chatme.controller;
+
+import android.util.Log;
+
+import com.google.codeu.chatme.model.Conversation;
+import com.google.codeu.chatme.model.Message;
+import com.google.codeu.chatme.model.User;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+
+@Deprecated
+/**
+ * Functions defined in this class are being moved to presenter classes to follow MVP
+ * design pattern
+ */
+public class Controller {
+
+    private static final String TAG = Controller.class.getName();
+
+    private static DatabaseReference mRootRef = FirebaseDatabase.getInstance().getReference();
+
+    /**
+     * Saves a new conversation to Firebase real-time database
+     *
+     * @param owner id of {@link Conversation} owner creator
+     */
+    public static void addConversation(String owner) {
+        Conversation newConversation = new Conversation(owner);
+
+        // generates unique id for conversation to be saved
+        final String newConversationId = mRootRef.child("conversations").push().getKey();
+
+        mRootRef.child("conversations").child(newConversationId).setValue(newConversation,
+                new DatabaseReference.CompletionListener() {
+
+                    @Override
+                    public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
+                        if (databaseError != null) {
+                            Log.w(TAG, "addConversation:failure " + databaseError.getMessage());
+                        } else {
+                            Log.i(TAG, "addConversation:success " + newConversationId);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Saves a new message to Firebase real-time database
+     *
+     * @param author         id of message author ({@link User})
+     * @param conversationId id of conversation the message is associated with
+     * @param content        string representation of message content
+     */
+    public static void addMessage(String author, final String conversationId, String content) {
```

Comment:
Future changes adding more fields to messages may be simpler if this method accepts a ``Message`` parameter instead of the arguments to pass to a Message constructor.